js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillPartitionsTab.tsx

```diff
@@ -107,98 +107,103 @@ export const BackfillPartitionsTab = ({backfill}: {backfill: BackfillDetailsBack
-          } else {
-            targeted = 1;
-            failed = asset.failed ? 1 : 0;
-            inProgress = asset.inProgress ? 1 : 0;
-            completed = asset.materialized ? 1 : 0;
-          }
-          return (
-            <tr
-              key={tokenForAssetKey(asset.assetKey)}
-              data-testid={testId(`backfill-asset-row-${tokenForAssetKey(asset.assetKey)}`)}
-            >
-              <td>
-                <Box flex={{direction: 'row', justifyContent: 'space-between'}}>
-                  <div>
-                    <ButtonLink
-                      onClick={() =>
-                        onShowAssetDetails(
-                          asset.assetKey,
-                          asset.__typename === 'AssetPartitionsStatusCounts',
-                        )
-                      }
-                    >
-                      {asset.assetKey.path.join('/')}
-                    </ButtonLink>
-                  </div>
-                  <div>
-                    <StatusBar
-                      targeted={targeted}
-                      inProgress={inProgress}
-                      completed={completed}
-                      failed={failed}
-                    />
-                  </div>
-                </Box>
-              </td>
-              {asset.__typename === 'AssetPartitionsStatusCounts' ? (
-                <>
-                  <td>{targeted}</td>
-                  <td>{inProgress}</td>
-                  <td>{completed}</td>
-                  <td>{failed}</td>
-                </>
-              ) : (
-                <>
-                  <td>-</td>
-                  <td>
-                    {inProgress ? (
-                      <Tag icon="spinner" intent="primary">
-                        In progress
-                      </Tag>
-                    ) : (
-                      '-'
-                    )}
-                  </td>
-                  <td>{completed ? <Tag intent="success">Completed</Tag> : '-'}</td>
-                  <td>{failed ? <Tag intent="danger">Failed</Tag> : '-'}</td>
-                </>
-              )}
-            </tr>
-          );
-        })}
-      </tbody>
-    </Table>
+    <Box
+      flex={{direction: 'column'}}
+      style={{flex: 1, position: 'relative', minHeight: 0, overflow: 'auto'}}
+    >
+      <Table>
+        <thead>
+          <tr>
+            <th style={{width: '50%'}}>Asset name</th>
+            <th>
+              <Link to={getRunsUrl('targeted')}>Partitions targeted</Link>
+            </th>
+            <th>
+              <Link to={getRunsUrl('inProgress')}>In progress</Link>
+            </th>
+            <th>
+              <Link to={getRunsUrl('complete')}>Completed</Link>
+            </th>
+            <th>
+              <Link to={getRunsUrl('failed')}>Failed</Link>
+            </th>
+          </tr>
+        </thead>
+        <tbody>
+          {backfill.assetBackfillData?.assetBackfillStatuses.map((asset) => {
```

Comment:
can we virtualize this table while e're here

Reply:
Hmm I think this UI is getting replaced ~soon by the design here: https://www.figma.com/design/rsRWKdmEpnyZrkNcZR7wcR/Dagster-UI-2024-Q2?node-id=299-18820&t=Ss17TSitz45bXWMc-0, so I'd been ignoring the perf issues on this one, but I just noticed that moved to the "explorations" tab of Figma, let me see if this is happening soon or not 👀

Reply:
Converted this to a virtualized table like all our newer ones, also added middle truncation to the asset keys for small viewports:

![image](https://github.com/dagster-io/dagster/assets/1037212/e7e93323-7a70-48e1-afa2-e0dcdf9801ce)